packages/react-native-session-replay/android/src/main/kotlin/com/datadog/reactnative/sessionreplay/ShadowNodeWrapper.kt

```diff
@@ -0,0 +1,64 @@
+/*
+ *
+ *  * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ *  * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ *  * Copyright 2016-Present Datadog, Inc.
+ *
+ */
+
+package com.datadog.reactnative.sessionreplay
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.reactnative.sessionreplay.utils.ReflectionUtils
+import com.facebook.react.bridge.ReactContext
+import com.facebook.react.uimanager.ReactShadowNode
+import com.facebook.react.uimanager.UIImplementation
+import com.facebook.react.uimanager.UIManagerModule
+import java.util.concurrent.CountDownLatch
+
+internal class ShadowNodeWrapper(
+    private val logger: InternalLogger,
+    private val reactContext: ReactContext,
+    private val uiManagerModule: UIManagerModule?,
+    private val reflectionUtils: ReflectionUtils = ReflectionUtils(logger)
+) {
+    private var shadowNode: ReactShadowNode<out ReactShadowNode<*>>? = null
+
+    internal fun initialize(viewId: Int): Boolean {
+        shadowNode = getShadowNode(viewId)
+        return shadowNode != null
+    }
+
+    internal fun getDeclaredShadowNodeField(fieldName: String): Any? {
+        return reflectionUtils.getDeclaredField(
+            shadowNode?.javaClass?.superclass as Class<*>,
```

Comment:
we don't need `superclass` here, because `ReflectionUtils.getDeclaredField` can handle traversal

```suggestion
            shadowNode?.javaClass as Class<*>,
```
Moreover, we this code is still unsafe, because it does unsafe cast. We can avoid this, if you add static method to create `ShadowNodeWrapper` instance unique for each view, returning either instance with non-null `shadowNode` or null.

Reply:
I've changed to build from static